daft/dataframe/dataframe.py

```diff
@@ -64,6 +67,8 @@
 
 ManyColumnsInputType = Union[ColumnInputType, Iterable[ColumnInputType]]
 
+OutputType = TypeVar("OutputType")
+P = ParamSpec("P")
```

Comment:
```suggestion
T = TypeVar("T")
P = ParamSpec("P")
```

Let's also move this under the `if TYPE_CHECKING` block